### PR TITLE
Update scripts for create_dist

### PR DESCRIPTION
**Pull request by RyanJarv**
These are various improvements/fixes I added while working on the CI builds.

    * Add mac signing id and keychain parameters
    * Add gclient_verbose option to sync
    * Allow util.js func options to be set with out explicit cwd
    * Use git -C option to specify repos
    
      Switching directories on windows can cause strange issues in
      the case of depot_tools assumingly because the shell will see
      the git in the current directory and run that instead of what is
      run elsewhere.
    
    * Fix setting references on the command line with yarn sync.
    
      Commander camelcases options with dashs which breaks when used with
      '--brave-core_ref'. This also renames the option to --brave_core_ref

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Test Plan:
Have been using in the new CI infrastructure.

Will likely need to come back and start adding some tests for ./lib/* (none currently).

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const build = (buildConfig = config.defaultBuildConfig, options) => {
     util.updateBranding()
   }
 
-  util.buildMuon()
+  util.buildMuon('brave')
```
Comment (bbondy) [inline lib/build.js]: pls pr a name change fro muon here too separately.

File: lib/config.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const Config = function () {
   this.resourcesDir = path.join(this.rootDir, 'resources')
   this.defaultGClientFile = path.join(this.rootDir, '.gclient')
   this.gClientFile = process.env.BRAVE_GCLIENT_FILE || this.defaultGClientFile
+  this.gClientVerbose = getNPMConfig(['gclient_verbose']) || false
```
Comment (bridiver) [inline lib/config.js]: do we want this as an npm config? Seem like it makes more sense as a command line option

File: scripts/sync.js [hunk 3]
```diff
@@ -9,43 +9,45 @@ const projectNames = config.projectNames.filter((project) => config.projects[pro
...
+  project = project.replace('-', '_')
   program.option('--' + project + '_ref <ref>', project + ' ref to checkout')
 })
 
 program.parse(process.argv)
 config.update(program)
 
 if (program.init || program.submodule_sync) {
-  util.submoduleSync()
+  util.submoduleSync({verbose: config.gClientVerbose})
```
Comment (bridiver) [inline scripts/sync.js]: submodule sync doesn't use gclient

File: lib/config.js (hunk 2 above)
Comment (bbondy) [reply lib/config.js]: can be done as a followup

File: scripts/sync.js [hunk 4]
```diff
@@ -9,43 +9,45 @@ const projectNames = config.projectNames.filter((project) => config.projects[pro
...
 }
 
 if (program.init) {
-  util.buildGClientConfig()
+  util.buildGClientConfig({verbose: config.gClientVerbose})
```
Comment (bridiver) [inline scripts/sync.js]: `verbose` would have no effect here. This just writes the gclient config file

File: lib/util.js [hunk 5]
```diff
@@ -70,20 +73,21 @@ const util = {
     fs.copySync(path.join(braveAppDir, 'strings'), path.join(chromeComponentsDir, 'strings'))
   },
 
-  buildMuon: (options = config.defaultOptions) => {
-    console.log('building brave...')
+  buildMuon: (target = 'brave', options = config.defaultOptions) => {
```
Comment (bridiver) [inline lib/util.js]: defaults should be specified in the config

File: scripts/commands.js [hunk 6]
```diff
@@ -35,12 +37,17 @@ program
...
 program
   .command('create_dist')
-  .option('--mac_signing_identifier', 'The identifier to use for signing')
+  .option('-C <build_dir>', 'build config (out/Debug, out/Release')
   .option('--target_arch <target_arch>', 'target architecture', 'x64')
+  .option('--mac_signing_identifier <id>', 'The identifier to use for signing')
+  .option('--mac_signing_keychain <keychain>', 'The identifier to use for signing', 'login')
   .option('--debug_build <debug_build>', 'keep debugging symbols')
   .option('--official_build <official_build>', 'force official build settings')
+  .option('--brave_google_api_key <brave_google_api_key>')
```
Comment (bridiver) [inline scripts/commands.js]: these should only be needed at build time and not packaging

File: lib/config.js (hunk 2 above)
Comment (bridiver) [reply lib/config.js]: I see, it's also a command line option below

File: lib/createDist.js [hunk 7]
```diff
@@ -3,19 +3,16 @@ const util = require('../lib/util')
 const path = require('path')
 const fs = require('fs-extra')
 
-const createDist = (options) => {
+const createDist = (buildConfig = config.defaultBuildConfig, options) => {
+  config.buildConfig = buildConfig
```
Comment (bridiver) [inline lib/createDist.js]: I'm not sure I understand this change. `config.update` should handle updating the internal config and I don't think it's a good idea to set `buildConfig` directly like this. What is the reason for this change?

File: lib/config.js (hunk 2 above)
Comment (RyanJarv) [reply lib/config.js]: Yep, meant to be a cmd line options. Just grabbing from the npm config to match the other flags.

File: scripts/commands.js (hunk 6 above)
Comment (RyanJarv) [reply scripts/commands.js]: Currently create_dist will run the build if it hasn't already

File: lib/createDist.js (hunk 7 above)
Comment (bridiver) [reply lib/createDist.js]: I see, it's just matching `build`

File: lib/util.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ const config = require('./config')
 const fs = require('fs-extra')
 
 const runGClient = (args, options = {}) => {
+  if (options.verbose) args.push('--verbose')
```
Comment (bridiver) [inline lib/util.js]: we might as well just access `config.gClientVerbose` directly here. If any other options are passed to `gclientSync` or `gclientRunhooks` then the verbose option will be lost the way it's used now

File: scripts/commands.js (hunk 6 above)
Comment (RyanJarv) [reply scripts/commands.js]: Yeah should still be able to do that, won't rebuild if you already ran `yarn run build`. But think it might make sense to expose the target in a more generic way so we don't have to make sure all the options stay in sync.

Comment (bridiver) [reply scripts/commands.js]: I think what might make sense here is to remove the call to `util.run('gn', ['gen', config.outputDir, '--args="' + args + '"'], cmdOptions)` in `createDist` if no options are specified. Then running `npm run create_dist` will use the existing args.gn instead of overwriting it
